viabel/optimizers.py

```diff
@@ -0,0 +1,808 @@
+import autograd.numpy as np
+import autograd.numpy.random as npr
+import autograd.scipy.stats.multivariate_normal as mvn
+import autograd.scipy.stats.t as t_dist
+from autograd.scipy.linalg import sqrtm
+
+from functools import partial
+import matplotlib.pyplot as plt
+
+import tqdm
+import scipy.stats as stats
+from  .optimization_diagnostics import autocorrelation, monte_carlo_se, monte_carlo_se2, compute_khat_iterates, \
+    gpdfit, montecarlo_se, compute_R_hat, compute_R_hat_halfway, compute_R_hat_halfway_light
+
+
+__all__ = [
```

Comment:
Please move adagrad_optimize to this file and update other references in the code and examples as necessary